### PR TITLE
feat(core): add container { defaultArgs, withDefaultArgs }

### PR DESCRIPTION
**Pull request by dolanor**

### CODE REVIEW

File: core/container.go
Comment (vito) [inline core/container.go:415]: nit: This is technically correct, but I think we can get away with just checking if `len(args) == 0`; it's really unusual to see a slice pointer in Go IMO, and there aren't any reasonable semantics to `exec(args: [])` anyway.

**Review by vito — APPROVED** [review]
This looks great, I have a tiny nitpick but I'm approving anyhow. Up to you.

Sorry for the slow review, had this among my sea of pinned tabs and lost track with the calls today. :sweat_smile:

**Comment by gerhard** [comment]
What needs to happen before this can be merged?

File: core/container.go
Comment (dolanor) [reply core/container.go:415]: That was part of the discussion with the `withoutArgs`.
I'll let like that, merge and create a dedicated issue to handle `withoutDefaultArgs` instead of nil args slice.

**Comment by dolanor** [comment]
Some update need to happen because of https://github.com/dagger/dagger/issues/3284

**Pull request merged**
